packages/flutter/test/material/refresh_indicator_test.dart

```diff
@@ -46,6 +46,48 @@ void main() {
     await tester.pump(const Duration(seconds: 1)); // finish the indicator hide animation
     expect(refreshCalled, true);
   });
+  
+  testWidgets('Refresh Indicator - nested', (WidgetTester tester) async {
+    refreshCalled = false;
+    await tester.pumpWidget(
+      new MaterialApp(
+        home: new RefreshIndicator(
+          acceptNotification: (notification) => notification.depth == 1,
+          onRefresh: refresh,
+          child: new SingleChildScrollView(
+            scrollDirection: Axis.horizontal,
+            child: new Container(
+              width: 600.0,
+              child:new ListView(
+                physics: const AlwaysScrollableScrollPhysics(),
+                children: <String>['A', 'B', 'C', 'D', 'E', 'F'].map((String item) {
+                  return new SizedBox(
+                    height: 200.0,
+                    child: new Text(item),
+                  );
+                }).toList(),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+    
+    await tester.fling(find.text('A'), const Offset(300.0, 0.0), 1000.0); // horizontal fling
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1)); // finish the scroll animation
+    await tester.pump(const Duration(seconds: 1)); // finish the indicator settle animation
+    await tester.pump(const Duration(seconds: 1)); // finish the indicator hide animation
+    expect(refreshCalled, false); 
+    
+
+    await tester.fling(find.text('A'), const Offset(0.0, 300.0), 1000.0); // vertical fling
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1)); // finish the scroll animation
+    await tester.pump(const Duration(seconds: 1)); // finish the indicator settle animation
+    await tester.pump(const Duration(seconds: 1)); // finish the indicator hide animation
+    expect(refreshCalled, true); 
+  });
```

Comment:
need a test for the overscroll indicator too

Reply:
Done